core/iwasm/aot/aot_runtime.c

```diff
@@ -2300,7 +2300,7 @@ invoke_wasm_c_api_native(AOTModuleInstance *module_inst, void *func_ptr,
     }
     if (trap) {
         if (trap->message->data) {
-            snprintf(fmt, sizeof(fmt), "%%.%us", (uint32)trap->message->size);
+            snprintf(fmt, sizeof(fmt), "%%.%"PRIu32"s", (uint32)trap->message->size);
```

Comment:
%u, %d, %i and %x are common formats for printf/snprintf function, are they not supported by arm-none-eabi-gcc compiler? If we change code here, does it mean that we need to use PRIu32/PRId32/PRIx32 related strings in future code, which is really not convenient for most of the developers.

Another issue is that some platform doesn't support these strings, e.g. some embedded OSs, if we change the code, we had better add related macros in core/shared/platform/include/platform_common.h, e.g.:
```C
typedef int64_t int64;

#ifndef PRI64_PREFIX
# if UINTPTR_MAX == UINT64_MAX
#  define PRI64_PREFIX    "l"
#  define PRIPTR_PREFIX   "l"
# else
#  define PRI64_PREFIX    "ll"
#  define PRIPTR_PREFIX
# endif
#endif

#ifndef PRId32
# define PRId32     "d"
#endif
#ifndef PRIi32
# define PRIi32     "i"
#endif
#ifndef PRIu32
# define PRIu32     "u"
#endif
#ifndef PRIx32
# define PRIx32     "x"
#endif
#ifndef PRIX32
# define PRIX32     "X"
#endif
#ifndef PRIx64
# define PRIx64     PRI64_PREFIX "x"
#endif
```

Reply:
>%u, %d, %i and %x are common formats for printf/snprintf function, are they not supported by arm-none-eabi-gcc compiler?

Of course, they are supported by arm-none-eabi-gcc.
However, our environment (i.e. NuttX) typedefs _uint32_t as follows.

```
typedef unsigned long _uint32_t;
```

That's why the format warnings happen.

>Another issue is that some platform doesn't support these strings, e.g. some embedded OSs,

In my understanding, those strings should be supported by the recent compilers because they are defined in C99.
However, some RTOS (e.g. Zephyr) define those types in a header file in the RTOS.

Reply:
Yes, we can change.